### PR TITLE
Allow empty strings for promo attributes and ledeCaption

### PR DESCRIPTION
**Pull request by nigelhanlon**
Small change to allow the API to pass empty strings.

### CODE REVIEW

File: models/article.js [hunk 1]
```diff
@@ -21,12 +21,12 @@ module.exports = base.keys({
   dek: joi.string().required(),
   body: joi.string().required(),
   // promo details
-  promoHed: joi.string().required(),
-  promoDek: joi.string().required(),
+  promoHed: joi.string().allow('').required(),
```
Comment (almostagile) [inline models/article.js]: I wonder if we should just define these as default empty string, _i.e._ `joi.string().allow('').default('')`

Comment (nigelhanlon) [reply models/article.js]: So long as that would ensure they are defined and present in the article then I don't see an issue with it.

Comment (nigelhanlon) [reply models/article.js]: Seems to work fine with the suggested change so updated PR.

File: models/article.js [hunk 2]
```diff
@@ -21,12 +21,12 @@ module.exports = base.keys({
...
   // promo details
-  promoHed: joi.string().required(),
-  promoDek: joi.string().required(),
+  promoHed: joi.string().allow('').default(''),
+  promoDek: joi.string().allow('').default(''),
   photosTout: joi.array().items(photoModel).min(1).max(1).required(),
   // lede details
   photosLede: joi.array().items(photoModel).min(0).max(1).required(),
-  ledeCaption: joi.string().required(),
+  ledeCaption: joi.string().allow('').required(),
```
Comment (almostagile) [inline models/article.js]: Missed one

Comment (nigelhanlon) [reply models/article.js]: Well spotted, thank you 👍

**Pull request merged**